openedx/features/course_experience/utils.py

```diff
@@ -156,7 +162,7 @@ def dates_banner_should_display(course_key, user):
     if not RELATIVE_DATES_FLAG.is_enabled(course_key):
         return False, False
 
-    if RELATIVE_DATES_DISABLE_RESET_FLAG.is_enabled(course_key):
+    if RELATIVE_DATES_DISABLE_RESET_FLAG.is_enabled(course_key) and not allow_warning:
```

Comment:
Do we still need this? Maybe it would be better to remove this check from this function and move it directly to the following places.
https://github.com/openedx/edx-platform/blob/3949c73b354707c80084da52b60f8983cc666e2f/common/djangoapps/util/views.py#L210
https://github.com/openedx/edx-platform/blob/0eb896254825296aafa48798996418784972448c/openedx/features/course_experience/api/v1/views.py#L79

As we want to show the banner anyway (i.e., regardless of the state of the `RELATIVE_DATES_DISABLE_RESET_FLAG`), it should be possible to use the original logic in this function. This way, we could get rid of the `allow_warning` kwarg.

Reply:
@DanielVZ96, I added this check in https://github.com/openedx/edx-platform/pull/32148 because we initially didn't want to show the alert at all. Now that we want to restore the banner, this gets much simpler. 
Two views you mentioned are endpoints that reset the deadlines (I explicitly listed them in my previous comment), so we wouldn't change their logic. We would only change the values of `missed_gated_content` and `missed_deadlines` to be accurate in the Course Home serializer and `render_xblock`. That said, I don't think it's a blocker in this PR, and we can skip it.